src/display/canvas.js

```diff
@@ -213,12 +207,11 @@ var CachedCanvases = (function CachedCanvasesClosure() {
       var canvasEntry;
       if (this.cache[id] !== undefined) {
         canvasEntry = this.cache[id];
-        canvasEntry.canvas.width = width;
-        canvasEntry.canvas.height = height;
+        this.canvasFactory.reset(canvasEntry, width, height);
```

Comment:
Are you sure that this actually works, since I believe that you need to use `canvasEntry.canvas` instead (compare with the the `destroy(...)` call below).